### PR TITLE
[1] feat(2763): execute different bookends on different build cluster. BREAKING CHANGE: change constructor arguments

### PR DESCRIPTION
**Pull request by y-oksaku**
## Context

Different bookends can be executed on different build cluster.

## Objective

This PR changes the allows bookends to be switched for each build cluster.

## References

- Issues
  - https://github.com/screwdriver-cd/screwdriver/issues/2763

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -99,14 +99,24 @@ class Bookend extends BookendInterface {
     /**
      * Constructs the list of modules used by bookends
      * @constructor
-     * @param {Object}  defaultModules      key->instantiated plugin for default plugins provided by screwdriver-cd
-     * @param {Array}   setup               List of module names, or objects { name, config } for instantiation to use in sd-setup
-     * @param {Array}   teardown            List of module names, or objects { name, config } for instantiation to use in sd-teardown
+     * @param {Object}  defaultModules            key->instantiated plugin for default plugins provided by screwdriver-cd
+     * @param {Object}  config                    Object keyed by cluster name with setup/teardown bookend in each.
```
Comment (itleigns) [inline index.js]:
```suggestion
     * @param {Object}  config                    Object keyed by cluster name with value setup/teardown bookend.
```

**Review by itleigns — APPROVED** [review]

**Review by yk634 — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -116,11 +126,14 @@ class Bookend extends BookendInterface {
      * @param  {PipelineModel}  o.pipeline  Pipeline model for the build
      * @param  {JobModel}       o.job       Job model for the build
      * @param  {Object}         o.build     Build configuration for the build (before creation)
+     * @param  {String}         clusterName Cluster name
```
Comment (tkyi) [inline index.js]: brackets for optional param
```suggestion
     * @param  {String}         [clusterName] Cluster name
```

File: index.js [hunk 3]
```diff
@@ -136,11 +149,14 @@ class Bookend extends BookendInterface {
...
```
Comment (tkyi) [inline index.js]: same comment as above, brackets for optional param
```suggestion
     * @param  {String}         clusterName Cluster name
```

**Review by tkyi — APPROVED** [review]
very minor jsdoc comments

**Comment by y-oksaku** [comment]
@tkyi 
Thank you, I fixed the comments.

**Review by tkyi — APPROVED** [review]

**Pull request merged**